### PR TITLE
Using np.linalg.multi_dot

### PR DESCRIPTION
**Pull request by maldil**
I believe, it is more succinct and effective to refactor code to use `np.linalg.multi_dot` rather than numerous np.dot routines. What do you think about this change that has practical value?

### CODE REVIEW

File: pyriemann/utils/tangentspace.py [hunk 1]
```diff
@@ -74,7 +74,7 @@ def untangent_space(T, Cref):
         triuc = np.triu(covmats[i], 1) / np.sqrt(2)
         covmats[i] = (np.diag(np.diag(covmats[i])) + triuc + triuc.T)
         covmats[i] = expm(covmats[i])
-        covmats[i] = np.dot(np.dot(C12, covmats[i]), C12)
+        covmats[i] = np.linalg.multi_dot([C12, covmats[i], C12])
```
[the remark this answers is not in the document]
Comment (agramfort) [reply pyriemann/utils/tangentspace.py:77]: yes it's relevant when matrices can have different shapes at run time but here I suspect there is one order of doing the dots that is always faster no? If we know what this order is then multi_dot is not necessary. Did you time/profile the improvement?

[the remark this answers is not in the document]
Comment (qbarthelemy) [reply pyriemann/utils/tangentspace.py:77]: Thx for your contribution to pyRiemann!
Can you add `A @ B @ C @ D` in your comparison on square matrices?

[the remark this answers is not in the document]
Comment (qbarthelemy) [reply pyriemann/utils/tangentspace.py:77]: I don't see the same too... (varying matrices sizes, with 100 repetitions)

issue_185

```
import timeit
import numpy as np
from matplotlib import pyplot as plt


dims = [5, 10, 20, 30, 40, 50, 60]
n_dims, n_reps = len(dims), 100


t_md = np.zeros((n_dims, n_reps))
t_d = np.zeros((n_dims, n_reps))
t_a = np.zeros((n_dims, n_reps))

for i, n_dim in enumerate(dims):

    A = np.random.random((n_dim, n_dim))
    B = np.random.random((n_dim, n_dim))
    C = np.random.random((n_dim, n_dim))

    for j in range(n_reps):

        t0 = timeit.default_timer()
        np.linalg.multi_dot([A, B, C])
        t_md[i, j] = timeit.default_timer() - t0

        t0 = timeit.default_timer()
        np.dot(np.dot(A, B), C)
        t_d[i, j] = timeit.default_timer() - t0

        t0 = timeit.default_timer()
        A @ B @ C
        t_a[i, j] = timeit.default_timer() - t0


fig, ax = plt.subplots()
ax.errorbar(dims, t_md.mean(axis=1), yerr=t_md.std(axis=1), label='multi_dot')
ax.errorbar(dims, t_d.mean(axis=1), yerr=t_d.std(axis=1), label='dot')
ax.errorbar(dims, t_a.mean(axis=1), yerr=t_a.std(axis=1), label='@')
ax.set(xlabel='Matrices dimension', ylabel='Time')
plt.legend(loc='upper left')
```

**Comment by qbarthelemy** [comment]
Most of the `np.dot` between 2D arrays have been already replaced by `@` in recent PRs.

**Pull request closed without merging**